src/lib/xof/xof.h

```diff
@@ -203,6 +215,14 @@ class BOTAN_PUBLIC_API(3, 2) XOF {
        * @param output  the span to be filled entirely with output bytes
        */
       virtual void generate_bytes(std::span<uint8_t> output) = 0;
+
+      /**
+       * Clear the XOF's internal state and allow for new input.
+       */
+      virtual void reset() = 0;
```

Comment:
TBH I do not understand the semantic difference between `reset` and `clear` here; when/why would a user choose one vs the other? I have a feeling `reset` should be protected and only used internally

Reply:
> I have a feeling reset should be protected and only used internally

That was exactly my intention. I wanted to separate the public interface (`clear()`, not virtual) from the internal one (`reset()` virtual). So that the base class can safely handle the `m_xof_started` state.

In the sub-classes I forgot to move the `reset()` overrides into the `private:` section, though. Maybe that caused your confusion?

Reply:
Sorry, I missed in the diff that `XOF::reset` was declared in the private block!